### ISSUE TITLE
Setting color of individual trajectories

### ISSUE THREAD

**Issue by hichew22** [philips-software/latrend#152]
Hello,

I was wondering if there is a way to set the color of the individual trajectories within the plot() function. I would like to set the color of each individual trajectory according to the cluster it has been assigned to.

For example, for this example in the vignette:
<img width="712" alt="image" src="https://github.com/philips-software/latrend/assets/137368713/5c1c3a9d-8f33-48ba-8f20-0bea2e4706d5">

All of the lines in A would be red, all the lines in B would be green, and so forth (rather than black).

Thank you!

**Comment by niekdt** [comment]
Hi, 

There's currently no option to do this directly, but I see some advantages to making your proposal the default. I'll look into this.

For now, you can achieve it relatively easily by using `plotTrajectories` without faceting (this colors the trajectories), then faceting manually, and adding black cluster trajectories on top.

To illustrate:
```R
data(latrendData)
method <- lcMethodLMKM(Y ~ Time, id = "Id", time = "Time")
model <- latrend(method, latrendData, nClusters = 3)
```
Now to create the plot:
```R
plotTrajectories(trajs, cluster = trajectoryAssignments(model), facet = FALSE) +
    facet_wrap(~ Cluster) + # enable facet by cluster
    geom_line(data = clusterTrajectories(model), aes(x = Time, y = Y), color = 'black') # add cluster lines
```

![image](https://github.com/philips-software/latrend/assets/8193083/e53b8f81-3de7-4552-a91f-499e3cd2251f)


Note that for drawing the cluster trajectories, you need to specify the correct time and value variables (Time and Y in the example).

**Comment by hichew22** [comment]
When I try to do this with my model, I get an error saying object 'Cluster' not found. Here is a reproducible example of my code using the latrend data instead of my data:

I fit the model and select the 4-cluster model as so:
`data(latrendData)
options(latrend.id = "Id", latrend.time = "Time")
kml_method <- lcMethodKML("Y", nClusters = 2, nbRedrawing = 1)
kml_model <- latrend(kml_method, data = latrendData)
kml_methods <- lcMethods(kml_method, nClusters = 1:5)
kml_models <- latrendBatch(kml_methods, data = latrendData)
kml_model_4 <- subset(kml_models, nClusters == 4, drop = TRUE))`

And then try to plot:
`plotTrajectories(
  kml_model_4,
  cluster = trajectoryAssignments(kml_model_4),
  size = 1,
  facet = FALSE
) +
  facet_wrap(~ Cluster) +
  geom_line(data = clusterTrajectories(kml_model_4),
            aes(x = time, y = Y),
            color = Cluster)`

**Comment by hichew22** [comment]
Thank you, this is working for me! Is there a way to change the alpha of the individual trajectories so that they are less transparent?

**Comment by hichew22** [comment]
I created a data frame of the IDs, Y, Time, and cluster assignments (df). I also have a dataframe of the cluster trajectories (df_clust_traj). 

I tried creating a ggplot as so:
`p_kml <- ggplot() +
  geom_line(
    data = df,
    aes(
    x = Time,
    y = Y,
    fill = ID,
    color = Cluster
  ),
  alpha = 0.3) +
  facet_wrap(~ Cluster) +
  geom_line(data = df_clust_traj,
            aes(x = Time, y = Y, color = Cluster))`

Now, there are 4 facets (1 for each Cluster), and the individual trajectories are plotted within each facet. However, each facet has all 4 cluster trajectories. For example, this is 1 of the facets:

<img width="325" alt="image" src="https://github.com/philips-software/latrend/assets/137368713/29e54787-ad68-472a-90a7-e3e2e381d4d5">


Can you help me edit my code so that each cluster trajectory is assigned to its respective facet?

**Comment by niekdt** [comment]
You were very close. This works for me:

```R
ggplot() + 
  geom_line( data = df, aes( x = Time, y = Y, group = Id, color = Cluster ), alpha = 0.3) + 
  facet_wrap(~ Cluster) + 
  geom_line(data = df_clust_traj, aes(x = Time, y = Y), color = 'black')
```

- I changed fill to group, because line geoms do not have a fill property, but we do want to draw each trajectory separately.
- I set the cluster lines to black so they are visible

I don't see each cluster trajectory per facet. Did you use:
```R
df = trajectories(kmlModel4)
df_clust_traj = clusterTrajectories(kmlModel4)
```

**Comment by hichew22** [comment]
When I use df = trajectories(kmlModel4), the dataframe only has ID, Time, and Y. So I had to extract the cluster assignments for each ID and join the dataframes so that df has ID, Time, Y, and Cluster:
`upn <- ids(kml_model_4)
cluster_kml <- trajectoryAssignments(kml_model_4)
df_cluster_kml <- cbind(upn, cluster_kml) %>%
  as.data.frame()

df <- df %>%
  left_join(., df_cluster_kml, by = "Id")`

I did use df_clust_traj = clusterTrajectories(kmlModel4). 

I'm hoping to generate a version of the plot with the cluster lines set to the same color as the individual trajectories. When I set color to 'Cluster' with the 2nd geom-line, then each facet shows all 4 cluster trajectories.

**Comment by niekdt** [comment]
Oops, I forgot to mention in my previous post that I made a change, reported in #153, where `trajectories()` now also outputs a Cluster column. Then you'll find that the code above works.

To get the change, install the latest commit from master using `remotes::install_github('philips-software/latrend')`

**Comment by hichew22** [comment]
Thank you! Is there a way to change the cluster trajectory within each facet to match the color of the individual trajectories? For example? Instead of black lines it would be red for cluster A, green for B, etc?

**Comment by niekdt** [comment]
Definitely, but then the cluster trajectories will be very hard to see against the trajectories.

Just set `color = Cluster ` as in your original code.

**Comment by hichew22** [comment]
When I do that, all 4 cluster trajectories show up in each facet. Does that occur for you?

**Comment by niekdt** [comment]
If you're using the original df code you posted, check if the cluster names are the same for both data.frames

This works for me:
```R
df = trajectories(kmlModel4)
df_clust_traj = clusterTrajectories(kmlModel4)

ggplot() + 
  geom_line( data = df, aes( x = Time, y = Y, group = Id, color = Cluster ), alpha = 0.3) + 
  facet_wrap(~ Cluster) + 
  geom_line(data = df_clust_traj, aes(x = Time, y = Y), color = 'black')
```

![image](https://github.com/philips-software/latrend/assets/8193083/c3eec6c2-37a4-401b-8a02-fddfb4843354)
